src/scripts/services/calculatePercentage.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+module.exports =
+angular.module("quill-grammar.services.calculatePercentage", [])
+.factory('calculatePercentageService', function () {
+  var calculatePercentage = function (conceptTagResults) {
+    var numCorrect, percentage;
+    if (conceptTagResults.length == 0) {
```

Comment:
the issue isnt when the length is undefined, its when the length is 0 (in which case the percentage calculating will involve a 0 in the denominator)